clang/lib/Analysis/FlowSensitive/HTMLLogger.cpp

```diff
@@ -430,11 +430,10 @@ class HTMLLogger : public Logger {
               AST.getSourceManager(), AST.getLangOpts());
           if (EltRange.isInvalid())
             continue;
-          if (EltRange.getBegin() < Range.getBegin() ||
-              EltRange.getEnd() >= Range.getEnd() ||
-              EltRange.getEnd() < Range.getBegin() ||
-              EltRange.getEnd() >= Range.getEnd())
+          if (EltRange.getEnd() <= Range.getBegin() ||
```

Comment:
This looks like it misses the case where `EltRange.getBegin() < Range.getBegin()`; shouldn't this be:
```
if (EltRange.getBegin() < Range.getBegin() ||
    EltRange.getEnd() >= Range.getEnd())
```